bittensor/_metagraph/metagraph_impl.py

```diff
@@ -275,116 +274,40 @@ def sync(self, force: bool = False ):
                 force (bool):
                     force syncs all nodes on the graph.
         """
-        loop = asyncio.get_event_loop()
-        loop.set_debug(enabled=True)
-        loop.run_until_complete(self._async_sync(force))
 
-    async def _async_sync( self, force: bool = False ):
-        r""" Uses the passed subtensor interface to update the metagraph chain state to reflect 
-            the latest info on chain.
-
-            Args: 
-                subtensor: (:obj:`bittensor.Subtensor`, optional):
-                    Subtensor chain interface obbject. If None, creates default connection to kusanagi.
-        """
         # Query chain info.
-        chain_lastemit = dict( await self.subtensor.async_get_last_emit() ) #  Optional[ List[Tuple[uid, lastemit]] ]
-        chain_stake = dict( await self.subtensor.async_get_stake() ) #  Optional[ List[Tuple[uid, stake]] ]
-        chain_block = int( await self.subtensor.async_get_current_block()) #  Optional[ int ]
+        chain_lastemit = dict( self.subtensor.get_last_emit() ) #  Optional[ List[Tuple[uid, lastemit]] ]
+        chain_stake = dict( self.subtensor.get_stake() ) #  Optional[ List[Tuple[uid, stake]] ]
+        chain_block = int( self.subtensor.get_current_block()) #  Optional[ int ]
+        chain_weights_uids = dict ( self.subtensor.get_weight_uids() )
+        chain_weights_vals = dict ( self.subtensor.get_weight_vals() )
+        chain_endpoints = dict ( self.subtensor.neurons() )
 
         # Build new state.
-        new_size = len(chain_stake)
-        old_size = self.n.item() 
-        old_block = self.block.item()
-        new_n = torch.tensor([new_size], dtype=torch.int64)
+        size = len(chain_stake)
+        new_n = torch.tensor([size], dtype=torch.int64)
         new_block = torch.tensor([chain_block], dtype=torch.int64)
-        new_uids = torch.tensor( range(new_size) ,  dtype=torch.int64)
-        new_stake = torch.tensor([ (float(chain_stake[uid])/1000000000) for uid in range(new_size)],  dtype=torch.float32)
-        new_lastemit = torch.tensor([ chain_lastemit[uid] for uid in range(new_size)], dtype=torch.int64)
+        new_uids = torch.tensor( range(size) ,  dtype=torch.int64)
 
         # Set params.
+        new_stake = torch.tensor([float(stake) / 1000000000 for stake in chain_stake], dtype=torch.float32)
+        new_lastemit = torch.tensor([float(emit) / 1000000000 for emit in chain_lastemit], dtype=torch.float32)
```

Comment:
Any reason to change the last emit from an int 64 to a float 32?

Reply:
@Eugene-hu he's dividing by 10^x so it would be a float not an integer either ways as opposed to when he is just appending the int to the list